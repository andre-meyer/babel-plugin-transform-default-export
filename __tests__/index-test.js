const babel = require('babel-core');
const plugin = require('../')

var defaultImport = `
import Box from '@material-ui/core/Box';
`;

var manyDefaultImports = `
import Box from '@material-ui/core/Box';
import SvgIcon from '@material-ui/core/SvgIcon';
import Modal from '@material-ui/core/Modal';

import RouterIcon from '@material-ui/icons/RouterIcon';
`;

var renamedDefaultImports = `
import Renamed from '@material-ui/styles/NotTheName';
`;

var nonDefaultImport = `
import { Box } from '@material-ui/core';
`;

var kindaDefaultMemberImport = `
import { makeStyles } from '@material-ui/core/styles';
`;

var multipleMemberImport = `
import { makeStyles, useTheme } from '@material-ui/core/styles';
`

var invalidMultipleMemberImport = `
import { amber, green } from '@material-ui/core/colors';
`

describe('default import transform', () => {
  it('transforms default import', () => {
    const {code} = babel.transform(defaultImport, {plugins: [plugin]});
    expect(code).toMatchSnapshot();
  });

  it('transforms many default imports', () => {
    const {code} = babel.transform(manyDefaultImports, {plugins: [plugin]});
    expect(code).toMatchSnapshot();
  })

  it('transform altering names from default import', () => {
    const {code} = babel.transform(renamedDefaultImports, {plugins: [plugin]});
    expect(code).toMatchSnapshot();
  })

  it('transforms memberstyle imports when nested', () => {
    const {code} = babel.transform(kindaDefaultMemberImport, {plugins: [plugin]});
    expect(code).toMatchSnapshot();
  })

  it('transforms multiple memberstyle imports when nested', () => {
    const {code} = babel.transform(multipleMemberImport, {plugins: [plugin]});
    expect(code).toMatchSnapshot();
  })

  it('transforms with special rules for unknown nested subpacke imports', () => {
    const {code} = babel.transform(invalidMultipleMemberImport, {plugins: [plugin]});
    expect(code).toMatchSnapshot();
  })

  it('does not transform member import', () => {
    const {code} = babel.transform(nonDefaultImport, {plugins: [plugin]});
    expect(code).toMatchSnapshot();
  })
})