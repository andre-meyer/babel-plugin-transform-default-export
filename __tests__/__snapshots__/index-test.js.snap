// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default import transform does not transform member import 1`] = `
"
import { Box } from '@material-ui/core';"
`;

exports[`default import transform transform altering names from default import 1`] = `"import { NotTheName as Renamed } from '@material-ui/styles';"`;

exports[`default import transform transforms default import 1`] = `"import { Box } from '@material-ui/core';"`;

exports[`default import transform transforms many default imports 1`] = `
"import { Box } from '@material-ui/core';
import { SvgIcon } from '@material-ui/core';
import { Modal } from '@material-ui/core';
import { RouterIcon } from '@material-ui/icons';"
`;

exports[`default import transform transforms memberstyle imports when nested 1`] = `"import { makeStyles } from '@material-ui/core';"`;

exports[`default import transform transforms multiple memberstyle imports when nested 1`] = `"import { makeStyles, useTheme } from '@material-ui/core';"`;

exports[`default import transform transforms with special rules for unknown nested subpacke imports 1`] = `
"import colors from '@material-ui/core';
const amber = colors.amber;
const green = colors.green;"
`;
